### PR TITLE
Refactor and update `scripts/add_*.py`

### PR DESCRIPTION
**Pull request by not-my-profile**

### CODE REVIEW

**Comment by sbrugman** [comment]
Nice work. I noticed that the scripts are not mentioned in `CONTRIBUTING.md`. Perhaps a good moment to add it there.

**Comment by charliermarsh** [comment]
Yeah I omitted them because they were so hacky :joy:

**Comment by not-my-profile** [comment]
They still are.^^ This PR just fixes the biggest problems but they could still be much improved ... but I am currently working on the many to many mapping between codes and rules.

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -84,6 +84,22 @@ jobs:
...
 
+  scripts:
+    name: "test scripts"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: nightly-2022-11-01
```
Comment (charliermarsh) [inline .github/workflows/ci.yaml:95]: I think this can probably be changed _not_ to use nightly.

Comment (not-my-profile) [reply .github/workflows/ci.yaml]: Good call! Changed :)

**Pull request merged**